jasperreports/pom.xml

```diff
@@ -0,0 +1,87 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
+                             http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <artifactId>struts-examples</artifactId>
+    <groupId>org.apache.struts</groupId>
+    <version>1.1.0</version>
+  </parent>
+
+  <artifactId>jasperreports</artifactId>
+  <packaging>war</packaging>
+
+  <name>JasperReports Tutorial</name>
+  <url>https://struts.apache.org/getting-started/jasper-reports-tutorial</url>
+  <properties>
+    <spring.platformVersion>5.3.20</spring.platformVersion>
+    <jasperreports.version>6.20.0</jasperreports.version>
+  </properties>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.springframework</groupId>
+        <artifactId>spring-framework-bom</artifactId>
```

Comment:
Firstly thank you very much for your PR!

Just IMHO add spring as dependency is too much for this sample I think. If you want then you might delete all spring stuff from your sample. Instead, in the beginning of action method, simply check if corresponding .jasper file exists or not. If exists then do nothing otherwise compile the .jrxml file and create the needed .jasper file.

This approach will have two benefits:

1. User can update and deploy a new report design without restarting app i.e. by deleting .jasper and .jrxml files and copying new .jrxml file.
2. The sample will be less complex and also will have less not required dependencies.

Reply:
Sorry @yasserzamani I was too rush :)

Reply:
Thanks for the code review.
I will edit the code again soon and submit a Pull Request again.

Reply:
> Sorry @yasserzamani I was too rush :)

No problem, the sample is already a nice jasperreport with spring sample :)

Reply:
If there are no major problems with the merged content, I will not make a new "Pull Request". 😄

I've only practiced in my personal Github repository.

* JasperReports example project with Spring

  * https://github.com/fp024/struts2-study/tree/master/jasper-reports-tutorial/jasper-reports-struts

* JasperReports example project without Spring - using Servlet's ContextListener

  * https://github.com/fp024/struts2-study/tree/master/jasper-reports-tutorial/jasper-reports-with-contextlistener

* JasperReports example project without Spring - using Interceptor

  * https://github.com/fp024/struts2-study/tree/master/jasper-reports-tutorial/jasper-reports-with-interceptor

  

  So have a nice day. 😄

Reply:
@fp024 no thank you, there's no major issue :) specially now that you've removed Spring bom. thanks again!